Packages/MIES/MIES_SweepFormula.ipf

```diff
@@ -1054,139 +1054,95 @@ static Function SF_GetDAChannel(string graph, variable sweep, variable channelTy
 	return NaN
 End
 
-static Function/WAVE SF_GetSweepForFormula(graph, range, channels, sweeps)
-	String graph
-	WAVE range, channels, sweeps
+static Function/WAVE SF_GetSweepForFormula(string graph, WAVE range, WAVE channels, WAVE sweeps)
 
-	variable i, j, rangeStart, rangeEnd, pOffset, delta, numRows, DAChannel
-	string dimLabel
-	variable channelType = -1
-	variable xStart = NaN, xEnd = NaN
+	variable i, j, rangeStart, rangeEnd, pOffset, delta, numRows, DAChannel, numSweeps, sweepNo
+	variable chanNr, chanType, numChannels, indexOffset, iStart, iLength
+	string dimLabel, device
 
 	ASSERT(WindowExists(graph), "graph window does not exist")
 	if(IsTextWave(range))
 		SF_ASSERT(DimSize(range, ROWS) == 1, "A epoch range must be a single string with the epoch name.")
 		WAVE/T wEpochName = range
 	else
-		SF_ASSERT(DimSize(range, ROWS) == 2, "A numerical range is of the form [rangeStart, rangeEnd].")
+		SF_ASSERT(DimSize(range, ROWS) == 2, "A numerical range is must have two rows for range start and end.")
 	endif
 	SF_ASSERT(DimSize(channels, COLS) == 2, "A channel input consists of [[channelType, channelNumber]+].")
 	SF_ASSERT(DimSize(sweeps, COLS) < 2, "Sweeps are one-dimensional.")
-	SF_ASSERT(DimSize(range, COLS) <= 1, "Multidimensional ranges not fully implemented.")
+	numSweeps = DimSize(sweeps, ROWS)
+	numChannels = DimSize(channels, ROWS)
 
-	WAVE/T/Z traces = GetTraceInfos(graph)
-	if(!WaveExists(traces))
-		DebugPrint("No traces found for extracting sweep wave locations.")
-		return $""
+	if(DimSize(range, COLS) > 0)
+		SF_ASSERT(DimSize(range, COLS) == numSweeps, "Multidimensional range column size must equal number of sweeps.")
+		SF_ASSERT(DimSize(range, LAYERS) == numChannels, "Multidimensional range layers size must equal number of channels.")
 	endif
 
-	SortColumns/A/DIML/KNDX={FindDimLabel(traces, COLS, "channelType"), FindDimLabel(traces, COLS, "channelNumber"), FindDimLabel(traces, COLS, "sweepNumber")} sortWaves=traces
+	SF_ASSERT(BSP_HasBoundDevice(graph), "No device bound.")
+	device = BSP_GetDevice(graph)
+	DFREF deviceDFR = GetDeviceDataPath(device)
```

Comment:
I bet this fail with a sweepbrowser started from the analysis browser. SB_UpdateSweepPlot has an example how to get a sweep datafolder from a sweep browser map index.

Reply:
This whole thing gets easy if we limit ourselves to unique sweep numbers in the sweep browser map.

Reply:
I hope the latest changes address this.

Reply:
Jeep, I'd say yes!